src/demo/renderer.js

```diff
@@ -1,91 +1,37 @@
 import 'babel-polyfill';
 import _ from 'lodash';
-import constants, {Modes, ClassType, strForClassType} from './constants';
+import constants, {ClassType} from './constants';
 import {FishBodyPart} from '../utils/fishData';
-import {setState} from './state';
+import {getState} from './state';
 
-const FISH_CANVAS_WIDTH = 300;
-const FISH_CANVAS_HEIGHT = 200;
-
-// Initialize the renderer once.
-// This will generate canvases with the fish collection.
-export function init(canvas) {
-  canvas.width = constants.canvasWidth;
-  canvas.height = constants.canvasHeight;
-  clearBackground();
-  const state = setState({
-    canvas,
-    ctx: canvas.getContext('2d')
-  });
-
-  switch (state.currentMode) {
-    case Modes.Training:
-      initTrainingScreen(state);
-      break;
-    case Modes.Predicting:
-      initPredictingScreen(state);
-      break;
-    case Modes.Pond:
-      initPondScreen(state);
-      break;
-    default:
-      console.error('Unrecognized mode passed to renderer init method.');
+export const drawBackground = imgPath => {
+  const canvas = getState().backgroundCanvas;
+  if (!canvas) {
+    return;
```

Comment:
Since we only call drawBackground on init, if this failed we wouldn't get another chance to draw it for a given mode.  But hopefully we never expect to land in any of these early exit conditions?

Reply:
yeah, i did that to match the current experience, but would like to refactor to something retryable (e.g., we set something like `state.backgroundDirty` and the renderer sets the background if `state.backgroundDirty`)